### ISSUE TITLE
Haxelib re-installs git libs despite saying they're already installed

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxelib#512]
After `haxelib install build.hxml`, it skips already installed libraries, unless those are git libs, in which case it installs them again:

```
You already have hxcpp version git installed.
Installing hxcpp from https://github.com/haxiomic/hxcpp branch: patch-3
Library hxcpp current version is now git
```

This can be slow when you just want to insure the dependencies are correct before running `haxe build.hxml`

I intend to PR a fix (no timeline however)

**Comment by TheDrawingCoder-Gamer** [comment]
Git you can technically never be sure it’s the correct version. iirc you can specify a branch/commit by doing `git:branch` as the version

**Comment by Simn** [comment]
This seems somewhat confusing and I wonder if this is mostly about the actual messages. While it's true that a git version might already be installed, it's not clear if that version needs an update or not, as pointed out by @TheDrawingCoder-Gamer.

The assumption when using git dependencies is that you want to use the latest commit, and this requires checking the repository accordingly, which I think might be mislabeled as "Installing" here.

I don't understand the branch comment because the same appears to be true if a branch is specified. The only situation where we shouldn't run any checks period is if a specific commit is set as the version, though I don't remember if this is even supported.

Please verify/clarify if this is actually a behavioral issue or just about the messaging!

**Comment by RealyUniqueName** [comment]
> a specific commit is set as the version, though I don't remember if this is even supported.

Specifying git commits is supported.

I think on update haxelib should do exactly what user specified when installing the library:

- If library was installed without any commit/branch/tag specified then
  1. switch to the default branch
  2. update to the latest commit in that branch
- If library was installed with a branch specified
  1. switch to that branch
  2. update to the latest commit in that branch
- If library was installed at specific commit or tag - switch back to that commit or tag.

In all cases haxelib should ask for confirmation if updating is going to actually change anything (reset local changes, switch commit etc)

**Comment by Simn** [comment]
I agree with everything except the last sentence. I don't like interactive CLIs because they make CIs hang. Also, we already say that we want to `haxelib update`, so I don't see why we should have to say it again.

**Comment by RealyUniqueName** [comment]
> why we should have to say it again.

For the same reason various editing software ask for confirmation before exiting if any changes were not saved yet :)
There could be changes made by the user, which they don't want to lose.

**Comment by Simn** [comment]
But in that case whatever git command we're employing is going to fail anyway.

**Comment by RealyUniqueName** [comment]
Indeed. At least we should ask if there are any uncommited changes.
